src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-notification-list/rgw-bucket-notification-list.component.ts

```diff
@@ -0,0 +1,98 @@
+import { TableComponent } from '~/app/shared/datatable/table/table.component';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { URLBuilderService } from '~/app/shared/services/url-builder.service';
+import { Bucket } from '../models/rgw-bucket';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+import { catchError, tap } from 'rxjs/operators';
+import { TopicConfiguration } from '~/app/shared/models/notification-configuration.model';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+
+const BASE_URL = 'rgw/bucket';
+@Component({
+  selector: 'cd-rgw-bucket-notification-list',
+  templateUrl: './rgw-bucket-notification-list.component.html',
+  styleUrl: './rgw-bucket-notification-list.component.scss',
+  providers: [{ provide: URLBuilderService, useValue: new URLBuilderService(BASE_URL) }]
+})
+export class RgwBucketNotificationListComponent extends ListWithDetails implements OnInit {
+  @Input() bucket: Bucket;
+  @Output() updateBucketDetails = new EventEmitter();
+  @ViewChild(TableComponent, { static: true })
+  table: TableComponent;
+  permission: Permission;
+  tableActions: CdTableAction[];
+  columns: CdTableColumn[] = [];
+  selection: CdTableSelection = new CdTableSelection();
+  notification$: Observable<TopicConfiguration[]>;
+  subject$ = new BehaviorSubject<TopicConfiguration[]>([]);
+  notificationList: TopicConfiguration;
+  modalRef: any;
+  @ViewChild('filterTpl', { static: true })
+  filterTpl: TemplateRef<any>;
+  constructor(
+    private rgwBucketService: RgwBucketService,
+    private authStorageService: AuthStorageService,
+    public actionLabels: ActionLabelsI18n
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    this.permission = this.authStorageService.getPermissions().rgw;
+    this.columns = [
+      {
+        name: $localize`Name`,
+        prop: 'Id',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Topic`,
+        prop: 'Topic',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Event`,
+        prop: 'Event',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Filter`,
+        prop: 'Filter',
+        flexGrow: 1,
+        cellTemplate: this.filterTpl
+      }
+    ];
+  }
+
+  loadNotification(context: CdTableFetchDataContext) {
+    this.notification$ = this.rgwBucketService.getBucketNotificationList(this.bucket.bucket).pipe(
+      tap((notification: TopicConfiguration) => {
+        this.notificationList = notification;
+      }),
+      catchError((error) => {
+        context.error(error);
+        return of(null);
+      })
+    );
```

Comment:
i see you have initialized a `subject` above but you are not using it? you can take proper use of it so that when you refresh the data (every 5s), the entire observable doesn't get w
```suggestion
    this.notification$ = this.subject.pipe(
      switchMap(() => 
        this.rgwBucketService.getBucketNotificationList(this.bucket.bucket).pipe(
          catchError((error) => {
            this.context.error(error);
            return of(null);
          })
        )
      )
    )
```

and you don't need to use `tap()` here as you are not doing anything with it.

also this logic needs to exist inside the `ngOnInit()` and every table refresh `fetchData()` should only trigger the subject update like `this.subject.next([])`. 

you can have a look at https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-group/cephfs-subvolume-group.component.ts#L153

Reply:
I’ve made the suggested changes